### DISCUSSION TITLE
Unable to install pyaudio

### DISCUSSION THREAD

**Discussion by Ak-Tools** [termux/termux-packages#7349] [Q&A]
Everyone knows that portaudio is not available in the main-repo. It is available in the pointless-repo.

So, i install it by.

curl -LO https://its-pointless.github.io/setup-pointless-repo.sh

bash setup-pointless-repo.sh

pkg install portaudio

And it installed successfully But,

When i tried to install pyaudio by:- pip install pyaudio. Then it throughs an error.

Given below:-

~ $ pip install pyaudio
Collecting pyaudio
  Using cached PyAudio-0.2.11.tar.gz (37 kB)
Building wheels for collected packages: pyaudio
  Building wheel for pyaudio (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /data/data/com.termux/files/usr/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/data/data/com.termux/files/usr/tmp/pip-install-g6si8cn_/pyaudio_1c5a4748201e4a72ae1242db3bf44241/setup.py'"'"'; __file__='"'"'/data/data/com.termux/files/usr/tmp/pip-install-g6si8cn_/pyaudio_1c5a4748201e4a72ae1242db3bf44241/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /data/data/com.termux/files/usr/tmp/pip-wheel-c7dnqgrh
       cwd: /data/data/com.termux/files/usr/tmp/pip-install-g6si8cn_/pyaudio_1c5a4748201e4a72ae1242db3bf44241/
  Complete output (16 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-aarch64-3.9
  copying src/pyaudio.py -> build/lib.linux-aarch64-3.9
  running build_ext
  building '_portaudio' extension
  creating build/temp.linux-aarch64-3.9
  creating build/temp.linux-aarch64-3.9/src
  aarch64-linux-android-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-strong -O3 -fstack-protector-strong -O3 -fPIC -I/data/data/com.termux/files/usr/include/python3.9 -c src/_portaudiomodule.c -o build/temp.linux-aarch64-3.9/src/_portaudiomodule.o
  src/_portaudiomodule.c:29:10: fatal error: 'portaudio.h' file not found
  #include "portaudio.h"
           ^~~~~~~~~~~~~
  1 error generated.
  error: command '/data/data/com.termux/files/usr/bin/aarch64-linux-android-clang' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for pyaudio
  Running setup.py clean for pyaudio
Failed to build pyaudio
Installing collected packages: pyaudio
    Running setup.py install for pyaudio ... error
    ERROR: Command errored out with exit status 1:
     command: /data/data/com.termux/files/usr/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/data/data/com.termux/files/usr/tmp/pip-install-g6si8cn_/pyaudio_1c5a4748201e4a72ae1242db3bf44241/setup.py'"'"'; __file__='"'"'/data/data/com.termux/files/usr/tmp/pip-install-g6si8cn_/pyaudio_1c5a4748201e4a72ae1242db3bf44241/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /data/data/com.termux/files/usr/tmp/pip-record-62alqrgk/install-record.txt --single-version-externally-managed --compile --install-headers /data/data/com.termux/files/usr/include/python3.9/pyaudio
         cwd: /data/data/com.termux/files/usr/tmp/pip-install-g6si8cn_/pyaudio_1c5a4748201e4a72ae1242db3bf44241/
    Complete output (16 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-aarch64-3.9
    copying src/pyaudio.py -> build/lib.linux-aarch64-3.9
    running build_ext
    building '_portaudio' extension
    creating build/temp.linux-aarch64-3.9
    creating build/temp.linux-aarch64-3.9/src
    aarch64-linux-android-clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -fstack-protector-strong -O3 -fstack-protector-strong -O3 -fPIC -I/data/data/com.termux/files/usr/include/python3.9 -c src/_portaudiomodule.c -o build/temp.linux-aarch64-3.9/src/_portaudiomodule.o
    src/_portaudiomodule.c:29:10: fatal error: 'portaudio.h' file not found
    #include "portaudio.h"
             ^~~~~~~~~~~~~
    1 error generated.
    error: command '/data/data/com.termux/files/usr/bin/aarch64-linux-android-clang' failed with exit code 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /data/data/com.termux/files/usr/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/data/data/com.termux/files/usr/tmp/pip-install-g6si8cn_/pyaudio_1c5a4748201e4a72ae1242db3bf44241/setup.py'"'"'; __file__='"'"'/data/data/com.termux/files/usr/tmp/pip-install-g6si8cn_/pyaudio_1c5a4748201e4a72ae1242db3bf44241/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /data/data/com.termux/files/usr/tmp/pip-record-62alqrgk/install-record.txt --single-version-externally-managed --compile --install-headers /data/data/com.termux/files/usr/include/python3.9/pyaudio Check the logs for full command output.

Plz help me to solve this.

**Comment by ar664** [comment]
portaudio is now in the main repo so installing it should be fine now.
I suggest retrying it if you haven't updated?
